src/containers/Home.js

```diff
@@ -3,16 +3,31 @@ import { connect } from "react-redux";
 import { withRouter, Link } from "react-router-dom";
 
 import { getMovies } from "../store/actions/MovieActions";
-import MovieCard from "../component/MovieCard";
+import MovieCard from "../component/movie-components/MovieCard";
 import Pagination from "react-js-pagination";
+import Search from "../component/movie-components/Search";
 
 class Home extends Component {
-  componentDidMount() {
-    const params = new URLSearchParams(this.props.location.search);
+  fetchMovies(title, pageNumber, elementsPerPage) {
     this.props.getMovies({
-      pageNumber: params.get("page") || 1,
-      elementsPerPage: params.get("elementsPerPage") || 5
+      pageNumber,
+      title,
+      elementsPerPage
     });
+    this.createHistory(title, pageNumber, elementsPerPage);
+  }
+  createHistory(title, pageNumber, elementsPerPage) {
```

Comment:
Add default values for `pageNumber` and `elementsPerPage` parameters to ensure it won't be `null` or `undefined`